src/devices/bus/hp80_io/82937.cpp

```diff
@@ -0,0 +1,354 @@
+// license:BSD-3-Clause
+// copyright-holders: F. Ulivi
+/*********************************************************************
+
+    82937.cpp
+
+    82937 module (HPIB interface)
+
+    TODO: Implement Parallel Poll response
+
+    Thanks to Tim Nye & Everett Kaser for dumping the 8049 ROM
+
+    Main reference for this module is:
+    HP 82937-90007, oct 80, HP82937A HP-IB Installation and theory
+    of operation manual
+
+*********************************************************************/
+
+#include "emu.h"
+#include "82937.h"
+#include "coreutil.h"
+
+// Debugging
+#define VERBOSE 0
+#include "logmacro.h"
+
+// Bit manipulation
+namespace {
+	static constexpr unsigned BIT_MASK(unsigned n)
+	{
+		return 1U << n;
+	}
+
+	template<typename T> void BIT_SET(T& w , unsigned n)
+	{
+		w |= (T)BIT_MASK(n);
+	}
+}
+
+// Bits in U3 (m_latch)
+static constexpr unsigned LATCH_CA_BIT = 5; // Controller Active
+static constexpr unsigned LATCH_TA_BIT = 4; // Talker Active
+static constexpr unsigned LATCH_EN_IFC_INT_BIT = 3; // Enable IFC interrupt
+static constexpr unsigned LATCH_EN_REN_INT_BIT = 2; // Enable REN interrupt
+static constexpr unsigned LATCH_EN_ATN_INT_BIT = 1; // Enable ATN interrupt
+static constexpr unsigned LATCH_EN_NDAC_BIT = 0;    // Enable NDAC
+
+// Bits on P1 port of 8049
+static constexpr unsigned P1_IFC_BIT = 7;
+static constexpr unsigned P1_REN_BIT = 6;
+static constexpr unsigned P1_SRQ_BIT = 5;
+static constexpr unsigned P1_ATN_BIT = 4;
+static constexpr unsigned P1_EOI_BIT = 3;
+static constexpr unsigned P1_DAV_BIT = 2;
+static constexpr unsigned P1_NDAC_BIT = 1;
+static constexpr unsigned P1_NRFD_BIT = 0;
```

Comment:
Qualifying these these `static` adds clutter:
* `constexpr` integers at global scope imply static (ODR only applies at compilation unit level)
* You could just put them in the anonymous namespace anyway to save the clutter of the keyword on every line.